src/graph.cc

```diff
@@ -174,11 +174,22 @@ bool DependencyScan::RecomputeOutputDirty(Edge* edge,
     }
 
     if (output_mtime < most_recent_input->mtime()) {
+      time_t output_t = static_cast<time_t>(output_mtime);
+      time_t most_recent_input_t = static_cast<time_t>(most_recent_input->mtime());
+
+      char *output_t_buff = ctime(&output_t);
+      char *most_recent_input_t_buff = ctime(&most_recent_input_t);
+
+      char output_t_str[100] = {0,};
+      char most_recent_input_t_str[100] = {0,};
+      strncpy(output_t_str, output_t_buff, strlen(output_t_buff) - 1);
```

Comment:
Oops, this is fragile. First we hardcode the length of `output_t_str[]` to 100, then we fill it with `strlen()` characters. How sure are we that we're not overflowing here?

I don't think the C standard says anything about the length of the static the function writes into. I think you should take the minimum between 100 (or 99) and that.